src/controller/python/chip/clusters/attribute.cpp

```diff
@@ -225,23 +225,25 @@ class ReadClientCallback : public ReadClient::Callback
 
     void OnReportEnd() override { gOnReportEndCallback(mAppContext); }
 
-    void OnDone(ReadClient *) override
+    void OnDone(ReadClient * apReadClient) override
     {
         gOnReadDoneCallback(mAppContext);
 
-        delete this;
+        // SubscriptionType will delete with pychip_ReadClient_Abort
+        if (apReadClient->IsReadType())
```

Comment:
onDone should take care of the readclient release. Abort is called from python application, and it is not called from IM's readClient, when subscription onError is called after backoff expired, onDone is further called, then memory leak for readClient would happen.

Reply:
In the original design, when a subscription `OnError` occurs, the memory is released in `OnDone`. If `pychip_ReadClient_Abort` is called again at this time, it will cause a secondary release and cause a crash.